src/main/java/org/folio/service/profiles/mappingprofile/MappingProfileServiceImpl.java

```diff
@@ -123,31 +123,52 @@
 
   @Override
   public Future<Void> validate(MappingProfile mappingProfile, OkapiConnectionParams params) {
+    Promise<Void> promise = Promise.promise();
     if (CollectionUtils.isNotEmpty(mappingProfile.getTransformations())) {
-      return transformationFieldsService.getTransformationFields(params).compose(transformationFieldCollection -> {
-        List<TransformationField> transformationFields = transformationFieldCollection.getTransformationFields();
-        for (Transformations transformation : mappingProfile.getTransformations()) {
+      transformationFieldsService.validateTransformations(mappingProfile.getTransformations())
+        .compose(v -> transformationFieldsService.getTransformationFields(params)
+          .compose(transformationFieldCollection -> validateTransformation(mappingProfile, transformationFieldCollection)))
```

Comment:
We should combine all validations to a single place, it's difficult to keep track of so many validate methods

Reply:
I think that this one was simplified well. Logic is the same. Wath do you think guys @illia-borysenko , @Igor-Gorchakov , @andrii-novytskyi ?

Reply:
we can create a separate class for validation purpose. @KVupp did you mean that?

Reply:
yea there are multiple validation methods and all over the place, we need to do the validation only in one place, move the other code along with it, so it's more readable

Reply:
We decided to handle in a future PR